lib/buffer.js

```diff
@@ -933,8 +940,12 @@ Buffer.prototype.readUInt32BE = function(offset, noAssert) {
 
 
 Buffer.prototype.readIntLE = function(offset, byteLength, noAssert) {
+  if (!noAssert)
+    checkByteLength(byteLength);
```

Comment:
@seppevs No matter the order of the assertions, I the typechecking part of `checkByteLength` should be consistent with `checkOffset`. If `checkOffset` doesn't check the type, `checkByteLength` should not do it either. If we do want to check the type, then we need to add this to `checkOffset` too.